clang/test/Driver/sycl-device-lib-nvptx.cpp

```diff
@@ -0,0 +1,40 @@
+// Tests specific to `-fsycl-targets=nvptx64-nvidia-nvptx`
+// Verify that the correct devicelib linking actions are spawned by the driver.
+// Check also if the correct warnings are generated.
+
+// UNSUPPORTED: system-windows
+
+// Check if internal libraries are still linked against when linkage of all device libs is manually excluded.
+// RUN: %clangxx -ccc-print-phases -std=c++11 \
+// RUN: -fsycl -fno-sycl-device-lib=all \
+// RUN: -fsycl-targets=nvptx64-nvidia-cuda %s 2>&1 \
+// RUN: | FileCheck -check-prefix=CHK-NO-DEVLIB %s
+
+// CHK-NO-DEVLIB: [[LIB1:[0-9]+]]: input, "{{.*}}libsycl-itt-user-wrappers.bc", ir, (device-sycl, sm_50)
```

Comment:
I would maybe also add some `CHK-NO-DEVICELIB-NOT`s to ensure the phases don't include linking with devicelib